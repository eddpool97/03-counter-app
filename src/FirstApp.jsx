import PropTypes from 'prop-types';

export const FirstApp = ({ title, message }) => {
  return (
    <>
      <h1>{title}</h1>
      <p>{message}</p>
      <p>{name}</p>
    </>
  );
};

//Manejo de PropTypes
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
};

//Manejo de DefaultProps
FirstApp.defaultProps = {
  title: 'No Value',
  message: 'No Value',
  name: 'Edwin Ayala'
};
