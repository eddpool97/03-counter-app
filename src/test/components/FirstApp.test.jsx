import { render } from "@testing-library/react";
import { FirstApp } from "../../FirstApp";


describe('Pruebas en <FirstApp />' ,()=>{
    test('Debe de hacer match con el snapshot', ()=>{
        const {container} = render(<FirstApp title="First App Running" message="Hello Edwin" name="Edwin Ayala"/>);

        expect( container ).toMatchSnapshot();

    });

    test('Debe de mostrar el titulo en H1', ()=>{

        const title = 'First App Running';

        const {container, getByText } = render(<FirstApp title="First App Running" message="Hello Edwin" name="Edwin Ayala"/>);

        expect( getByText(title) ).toBeTruthy();

        const h1 = container.querySelector('h1');
        console.log( h1.innerHTML );

        expect( container ).toMatchSnapshot();
    });
});