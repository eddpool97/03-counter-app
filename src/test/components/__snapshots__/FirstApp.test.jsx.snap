// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    First App Running
  </h1>
  <p>
    Hello Edwin
  </p>
  <p />
</div>
`;

exports[`Pruebas en <FirstApp /> Debe de mostrar el titulo en H1 1`] = `
<div>
  <h1>
    First App Running
  </h1>
  <p>
    Hello Edwin
  </p>
  <p />
</div>
`;
